test/integration/viewer_spec.mjs

```diff
@@ -87,4 +92,127 @@ describe("PDF viewer", () => {
       );
     });
   });
+
+  describe("CSS-only zoom", () => {
+    let pages;
+
+    beforeAll(async () => {
+      pages = await loadAndWait(
+        "tracemonkey.pdf",
+        ".textLayer .endOfContent",
+        null,
+        null,
+        {
+          maxCanvasPixels: 0,
+        }
+      );
+    });
+
+    afterAll(async () => {
+      await closePages(pages);
+    });
+
+    beforeEach(() =>
+      Promise.all(
+        pages.map(async ([browserName, page]) => {
+          await page.evaluate(() => {
+            window.PDFViewerApplication.pdfViewer.currentScaleValue = 1;
+
+            window.pageRenderedListener = new AbortController();
+            window.firstPageRenders = 0;
+            window.PDFViewerApplication.eventBus.on(
+              "pagerendered",
+              ({ pageNumber }) => {
+                if (pageNumber === 1) {
+                  window.firstPageRenders++;
+                }
+              },
+              { signal: window.pageRenderedListener.signal }
+            );
+          });
+        })
+      )
+    );
+
+    afterEach(() =>
+      Promise.all(
+        pages.map(async ([browserName, page]) => {
+          await page.evaluate(() => {
+            window.pageRenderedListener.abort();
+          });
+        })
+      )
+    );
+
+    function createPromiseForFirstPageRendered(page) {
+      return createPromise(page, (resolve, reject) => {
+        const controller = new AbortController();
+        window.PDFViewerApplication.eventBus.on(
+          "pagerendered",
+          ({ pageNumber }) => {
+            if (pageNumber === 1) {
+              resolve();
+              controller.abort();
+            }
+          },
+          { signal: controller.signal }
+        );
+        setTimeout(reject, 1000, "Timeout");
```

Comment:
Won't this *potentially* lead to a Promise being rejected with a `"Timeout"`-string?

If so please wrap that in an Error, i.e. `new Error("Timeout")`, since we generally don't allow rejecting with "raw" strings and there's even lint rules to enforce that. (But this case is probably special enough that those rules cannot detect it.)